ninja-servlet/src/main/java/ninja/servlet/NinjaBootstrap.java

```diff
@@ -191,7 +206,15 @@ protected void configureServlets() {
             return injector;
 
         } catch (Exception exception) {
-            logger.error("Fatal error booting Ninja", exception);
+            
+            // save exception in case other consumers of this class would like
+            // to handle an injector exception a little differently
+            injectorException = exception;
```

Comment:
If you can rewrite the initInjector to throw exceptions in a better way go ahead!

Btw replay: Yes - guice can be pretty terrible when it comes to error messages. That's one of the value propositions of dagger - but there are cons for dagger as well... So guice is quite okay right now :)